runtime/src/bank/builtins/core_bpf_migration/mod.rs

```diff
@@ -285,6 +285,101 @@ impl Bank {
 
         Ok(())
     }
+
+    /// Upgrade a Core BPF program.
+    /// To use this function, add a feature-gated callsite to bank's
+    /// `apply_feature_activations` function, similar to below.
+    ///
+    /// ```ignore
+    /// if new_feature_activations.contains(&feature_set::test_upgrade_program::id()) {
+    ///     self.upgrade_core_bpf_program(
+    ///        &core_bpf_program_address,
+    ///        &source_buffer_address,
+    ///        "test_upgrade_core_bpf_program",
+    ///     );
+    /// }
+    /// ```
+    #[allow(dead_code)] // Only used when an upgrade is configured.
+    pub(crate) fn upgrade_core_bpf_program(
+        &mut self,
+        core_bpf_program_address: &Pubkey,
+        source_buffer_address: &Pubkey,
+        datapoint_name: &'static str,
+    ) -> Result<(), CoreBpfMigrationError> {
+        datapoint_info!(datapoint_name, ("slot", self.slot, i64));
+
+        let target = TargetCoreBpf::new_checked(self, core_bpf_program_address)?;
+        let source = SourceBuffer::new_checked(self, source_buffer_address)?;
+
+        // Attempt serialization first before modifying the bank.
+        let new_target_program_account =
+            self.new_target_program_account(&target.program_address)?;
```

Comment:
Is this needed? Seems like there actually shouldn't be any modification of the program account.
...which means we can simplify some of the lamports calculations below as well, since `target.program_account.data().len()` will not change.

Reply:
Yeah, good call, we can probably leave the program account unchanged!